### PR TITLE
Sync unresolved packages from provenance checker report

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

We implement similar logic present for adviser sync. This is important if for some reason unsolved messages sent by provenance checker might not be consumed or they are lost.

Related-To: https://github.com/thoth-station/thoth-application/issues/818#issuecomment-770759067

Tested using: `provenance-checker-210202075859-fc13440741027d96`

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [x] Yes
- [ ] No

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -5352,6 +5352,28 @@ def sync_provenance_checker_result(self, document: dict) -> None:
...
             )
 
+            # Mark down packages that were not solved and missed in provenance checker.
+
+            report = document["result"]["report"]
+
+            message_id = "MISSING-PACKAGE"
+
+            for package in report:
+
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:5362]:
```suggestion
```

File: thoth/storages/graph/postgres.py [hunk 2]
```diff
@@ -5352,6 +5352,28 @@ def sync_provenance_checker_result(self, document: dict) -> None:
...
+                # Discover unresolved package matching the message ID
+                if package["id"] == message_id:
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:5364]:
```suggestion
                if package["id"] == "MISSING-PACKAGE":
```

File: thoth/storages/graph/postgres.py [hunk 3]
```diff
@@ -5352,6 +5352,28 @@ def sync_provenance_checker_result(self, document: dict) -> None:
                 user_software_stack_id=software_stack.id,
             )
 
+            # Mark down packages that were not solved and missed in provenance checker.
+
+            report = document["result"]["report"]
+
+            message_id = "MISSING-PACKAGE"
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:5359]:
```suggestion
```

**Review by fridex — COMMENTED** [review]
LGTM (just the blank lines could be deleted) :+1:

**Review by fridex — APPROVED** [review]
lgtm, thanks! :100: :beers:

**Comment by pacospace** [comment]
Based on https://github.com/thoth-station/thoth-application/issues/747, we can merge it for now!

**Pull request merged**